src/languages/sql.js

```diff
@@ -1,168 +1,647 @@
 /*
  Language: SQL
- Contributors: Nikolay Lisienko <info@neor.ru>, Heiko August <post@auge8472.de>, Travis Odom <travis.a.odom@gmail.com>, Vadimtro <vadimtro@yahoo.com>, Benjamin Auder <benjamin.auder@gmail.com>
  Website: https://en.wikipedia.org/wiki/SQL
- Category: common
+ Category: common, database
  */
 
+/*
+
+Goals:
+
+SQL is intended to highlight basic/common SQL keywords and expressions
+
+- If pretty much every single SQL server includes supports, then it's a canidate.
+- It is NOT intended to include tons of vendor specific keywords (Oracle, MySQL,
+  PostgreSQL) although the list of data types is purposely a bit more expansive.
+- For more specific SQL grammars please see:
+  - PostgreSQL and PL/pgSQL - core
+  - T-SQL - https://github.com/highlightjs/highlightjs-tsql
+  - sql_more (core)
+
+ */
+
+import * as regex from '../lib/regex.js';
+
 export default function(hljs) {
-  var COMMENT_MODE = hljs.COMMENT('--', '$');
+  const COMMENT_MODE = hljs.COMMENT('--', '$');
+  const STRING = {
+    className: 'string',
+    variants: [
+    {
+      begin: '\'', end: '\'',
+      contains: [{begin: '\'\''}]
+    },
+    {
+      begin: '"', end: '"',
+      contains: [{begin: '""'}]
+    },
+    {
+      begin: '`', end: '`'
+    }
```

Comment:
Backticks don't have any special meaning in SQL.

Reply:
Again, this isn't pure SQL... the idea is to also support common things (variables, etc) that are easy to support (for  a slightly more well rounded SQL grammar) IF those things don't cause problems across different engines.  So if backticks are strings in some engines and never used in others, this is a nice to have.

The problem comes with like `#` comments, which mean one thing in MySQL and something else in Oracle, etc... so they can't be part of this grammar, but I think backticks can.

Reply:
I really can see no point in this.
Following this logic, you can easily justify addition of _some_ non-standard keywords, popular in _some_ db engines. It doesn't cause problems, ugh?
And this takes us back to our discussion: https://github.com/highlightjs/highlight.js/pull/2740#issuecomment-706697239

Reply:
> I really can see no point in this.

The point is a grammar that "just works" in more situations.  A simple rule to handle quoting (if there are no other conflicts) is not at all the same as the murky swamp that is adding random keywords (since they cause real problems with auto-detect, WRONG highlighting, etc).  If one popular DB uses backtick for quoting but no others then I don't see the harm is supporting it (though perhaps we should add a comment).

See my comment at the top of the file:

```
SQL is intended to highlight basic/common SQL keywords and expressions

- If pretty much every single SQL server includes supports, then it's a canidate.
- It is NOT intended to include tons of vendor specific keywords (Oracle, MySQL,
  PostgreSQL) although the list of data types is purposely a bit more expansive.
```

The PRIMARY goal was to simplify the grammar and get us away from the unmaintained swamp of vendor specific keywords AND _vendor specific features that cause conflicts across different SQL engines._

The goal never was to draw a hard line in the sand and say "only the SQL standard, period".  Having an expanded list of datatypes makes this more useful for highlighting CREATE TABLE across multiple databases... but datatypes is scope a lot tighter than "any and all keywords", so I'm ok if it includes VARCHAR2, etc...

Reply:
And again, as soon you add `varchar2` to the list of datatypes, someone from Oracle world will come crying "Hey, I still see varchar2, so where are all the other Oracle stuff you had"?
What is the benefit of having `varchar2` but not `prior` or `rownum`?

Reply:
If you insist I'll remove it from v1 of this, but I'm much more in favor of "causes no harm" rather than "exactly the SQL spec only, period".  Long lists of unmaintainable keywords do cause harm and muddle the auto-detect.

Reply:
I rather try to reason. Sure you can leave it, but be ready to answer questions...

Reply:
I removed it.  Less is more.  If someone argues for its' inclusing in the future then we'll have more information at that time and can revisit.